src/App.tsx

```diff
@@ -1,30 +1,169 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { FormEvent, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import cn from 'classnames';
+
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodoStatus,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+
+export enum FilterTypes {
+  All,
+  Active,
+  Completed,
+}
+
+export enum ErrorMessage {
+  none = '',
+  load = 'Unable to load todos',
+  emptyTitle = 'Title should not be empty',
+  add = 'Unable to add a todo',
+  delete = 'Unable to delete a todo',
+  update = 'Unable to update a todo',
+}
+
+function filteringTodos(array: Todo[], filterType: FilterTypes) {
+  let arrayCopy = [...array];
+
+  switch (filterType) {
+    case FilterTypes.Active:
+      arrayCopy = arrayCopy.filter(item => !item.completed);
+      break;
+    case FilterTypes.Completed:
+      arrayCopy = arrayCopy.filter(item => item.completed);
+      break;
+    case FilterTypes.All:
+      return array;
+  }
+
+  return arrayCopy;
+}
```

Comment:
- what `array` is it? array/element/object/arr/obj/el/res/result are bad name and not informative at all
- You don't need to create a copy, `filter` doesn't mutate an array